src/models/plateau/parser.rs

```diff
@@ -0,0 +1,87 @@
+use super::{Cell, Plateau, EMPTY, PLAYER1, PLAYER2};
+use std::convert::TryFrom;
+
+impl TryFrom<String> for Plateau {
+	type Error = String;
+
+	fn try_from(map: String) -> Result<Self, String> {
+		if map.is_empty() {
+			return Err(String::from("Map is empty"));
+		}
+		let width = match map.find(|c| c == '\n') {
+			Some(w) => w,
+			none => map.len(),
+		};
+
+		let mut height = 0;
+		let mut cells = Vec::new();
+		for (y, row) in map.lines().enumerate() {
+			if row.len() != width {
+				return Err(format!("Rows sizes are inconsistant"));
```

Comment:
Doing a simple "String::from()" here might be better if the string won't be formatted

Reply:
Also "inconsistent"

Reply:
Fixed